pkg/apis/controlplane/types.go

```diff
@@ -280,6 +283,12 @@ type NetworkPolicyStats struct {
 	NetworkPolicy NetworkPolicyReference
 	// The stats of the NetworkPolicy.
 	TrafficStats statsv1alpha1.TrafficStats
+	// The stats of the NetworkPolicy, from rule perspective
+	RuleTrafficStats map[string]statsv1alpha1.TrafficStats
```

Comment:
The map is getting used in `func addRulesUp(ruleStats *[]statsv1alpha1.RuleTrafficStats, ruleSumStats *statsv1alpha1.TrafficStats, inc map[string]statsv1alpha1.TrafficStats)` as `inc` and  inside this function, using the map can help identify whether the `inc` collected from summary has already existed in the current ruleStats.

Anyway, the dilemma I am facing is trying to merging the `inc` from `nodestatssummary` to the current `ruleStats` with the same rule name. Using map reduces overhead of conversion from `statsv1alpha1.RuleTrafficStats` to `map[string]statsv1alpha1.TrafficStats` because as I mentioned above, we need to find whether the `inc` collected from summary has already existed in the current `ruleStats`.

Reply:
1. Then what is L289-L292 used for?
2. To keep efficient in addRulesUp, you can just convert slice to map, which is 3-4 lines of code. I think all current Antrea and Kubernetes APIs are following the convention I pasted above and some logic would prefer the items are stored in map for same requirement too but still did their own conversion. Maybe you could take a look at the doc and the discussions it linked to find the reasons why it's made so. And maybe in the end the reasons don't really matter for our cases, but it's still better to keep same style API for consistancy.

Reply:
1. L289-L292 are useless, I deleted them
2. I checked this discussion above(https://github.com/kubernetes/kubernetes/issues/2004). Basically there are two reasons to follow the convention: 1. clearer config file 2. to parse yaml tree easier 

Anyway I changed to code to follow the convention, thanks